service/frontend/workflow_handler.go

```diff
@@ -4743,6 +4780,31 @@ func (wh *WorkflowHandler) cancelOutstandingPoll(ctx context.Context, namespaceI
 	return true
 }
 
+func (wh *WorkflowHandler) registerOutstandingPollContext(
+	ctx context.Context,
+	pollerID string,
+	namespaceID string,
+) context.Context {
+
+	if pollerID != "" {
+		childCtx, cancel := context.WithCancel(ctx)
+		nsPollers, _ := wh.outstandingPollers.GetOrSet(namespaceID, collection.NewSyncMap[string, context.CancelFunc]())
```

Comment:
hmm, I don't like the idea that it's creating a map and throwing it out almost all the time. maybe it should pass in a factory function instead?

Reply:
changed to call get first then getorset.